explainer.md

```diff
@@ -0,0 +1,207 @@
+
+The table below shows the current state for the cross-section of these cases:
+
+* ✅ are cases which are currently customizable.
+* ☑ are cases which should be customizable but require new platform APIs.
+* ❌ are cases which should not be customizable
+
+|   | Same-Document  |  Cross-Document Same-Origin | Cross-Origin  |
+|---|---|---|---|
+| Swipe Navigation | ☑ | ☑ | ❌ |
+| Atomic Navigation  | ✅ | ☑ | ❌ |
+
+The platform APIs required for cases marked with ☑ are:
+
+* Gesture API: Most platforms have a swipe gesture which is used by the UA to trigger a back/forward navigation. On Android/iOS its swiping from the screen edge; on Mac, its multi-finger swipe. Site authors currently can not consistently intercept these events. Customizing swipe navigations for both same-document and cross-document navigations requires this[^1].
+
+* Persisting Visual State: The transition requires showing visual state from both before/after the navigation. Site authors can do this for same-document navigations but not cross-document. [ViewTransition](https://github.com/WICG/view-transitions/blob/main/explainer.md#cross-document-same-origin-transitions) addresses this gap.
+
+The exact details for the above APIs is out of scope for this proposal. These are meant to show that there are cases which are currently not customizable by site authors but will be going forward. The API proposals here should work well if/when one of these cases becomes customizable.
+
+# Problem Statement
+There is currently no way for a site to indicate that it has a custom visual transition tied to a navigation. For example, if the UA adds a visual transition for an atomic navigation (like clicking the back button), there is no way for the UA and site-author to coordinate the 2 transitions. The results in 2 category of issues:
+
+* Double transitions: Executing both the UA and author defined visual transitions looks like a visual glitch. This [site](https://darkened-relieved-azimuth.glitch.me) shows the issue on Chrome/Safari iOS and Safari on Mac. The site does a cross-fade when the user goes back:
+
+   * If the user navigates using the back button in the UI, a cross-fade defined by the site executes and is the only transition.
+
+   * If the user navigates using a swipe from the edge (iOS) or 2 finger swipe (mac), the UA does a transition showing content of the back entry. The Document then receives a `popState` event for the navigation and performs another visual transition.
+   
+   This has *compat risk* if the UA adds visual transitions for navigations which can be implicitly customized by the site author, i.e., same-document navigations. This is not a problem for cross-document navigations since customization requires UA primitives like ViewTransitions. The UA can use this to detect if there are custom transitions and give them precedence.
```

Comment:
```suggestion
   This has *compat risk* if the UA adds visual transitions for navigations which can be implicitly customized by the site author, i.e., same-document navigations. This is not a problem for cross-document navigations since customization requires UA primitives like [ViewTransitions](https://drafts.csswg.org/css-view-transitions-1/). The UA can use this to detect if there are custom transitions and give them precedence.
```